rpcs3/Emu/RSX/GL/GLTexture.cpp

```diff
@@ -687,14 +699,22 @@ namespace gl
 				if (driver_caps.ARB_compute_shader_supported)
 				{
 					u64 row_pitch = rsx::align2<u64, u64>(layout.width_in_block * block_size_in_bytes, caps.alignment);
-					if (!rsx::is_compressed_host_format(format))
-					{
-						// Handle emulated compressed formats with host unpack (R8G8 compressed)
-						row_pitch = std::max<u64>(row_pitch, dst->pitch());
-					}
+
+					// We're in the "else" branch, so "is_compressed_host_format()" is always false.
+					// Handle emulated compressed formats with host unpack (R8G8 compressed)
+					row_pitch = std::max<u64>(row_pitch, dst->pitch());
 
 					image_linear_size = row_pitch * layout.height_in_block * layout.depth;
 
+#if defined(ARCH_ARM64)
+					if (format == CELL_GCM_TEXTURE_COMPRESSED_DXT1 || format == CELL_GCM_TEXTURE_COMPRESSED_DXT23 || format == CELL_GCM_TEXTURE_COMPRESSED_DXT45)
+					{
+						// FIXME: The calculation above is incorrect; for this specific case, we need to use height_in_texel instead of height_in_block.
```

Comment:
This is only false because for compressed textures, block height is a divisor. Since this path is supposed to be uncompressed only, height_in_block == height_in_texel.
Using is_compressed_host_format() is probably the way to go here as you said.

Reply:
Thanks, I'll change accordingly.